frontend/src/scenes/heatmaps/heatmapsBrowserLogic.ts

```diff
@@ -0,0 +1,187 @@
+import { actions, afterMount, connect, kea, listeners, path, props, reducers, selectors } from 'kea'
+import { loaders } from 'kea-loaders'
+import api from 'lib/api'
+import { authorizedUrlListLogic, AuthorizedUrlListType } from 'lib/components/AuthorizedUrlList/authorizedUrlListLogic'
+import { RefObject } from 'react'
+
+import { HogQLQuery, NodeKind } from '~/queries/schema'
+import { hogql } from '~/queries/utils'
+import { PostHogAppToolbarEvent } from '~/toolbar/bar/toolbarLogic'
+
+import type { heatmapsBrowserLogicType } from './heatmapsBrowserLogicType'
+
+export type HeatmapsBrowserLogicProps = {
+    iframeRef: RefObject<HTMLIFrameElement | null>
+}
+
+export const heatmapsBrowserLogic = kea<heatmapsBrowserLogicType>([
+    path(['scenes', 'heatmaps', 'heatmapsBrowserLogic']),
+    props({} as HeatmapsBrowserLogicProps),
+
+    connect({
+        values: [
+            authorizedUrlListLogic({ actionId: null, type: AuthorizedUrlListType.TOOLBAR_URLS }),
+            ['urlsKeyed', 'checkUrlIsAuthorized'],
+        ],
+    }),
+
+    actions({
+        setBrowserSearch: (searchTerm: string) => ({ searchTerm }),
+        setBrowserUrl: (url: string) => ({ url }),
+        setIframePosthogJsConnected: (ready: boolean) => ({ ready }),
+        onIframeLoad: true,
+        onIframeToolbarLoad: true,
+        sendToolbarMessage: (type: PostHogAppToolbarEvent, payload?: Record<string, any>) => ({
+            type,
+            payload,
+        }),
+        setLoading: (loading: boolean) => ({ loading }),
+        loadTopUrls: true,
+        maybeLoadTopUrls: true,
+    }),
+
+    loaders({
+        browserSearchOptions: [
+            null as string[] | null,
+            {
+                setBrowserSearch: async ({ searchTerm }) => {
+                    const query: HogQLQuery = {
+                        kind: NodeKind.HogQLQuery,
+                        query: hogql`SELECT distinct properties.$current_url AS urls
+                                FROM events
+                                WHERE timestamp >= now() - INTERVAL 7 DAY
+                                AND timestamp <= now()
+                                AND properties.$current_url like '%${hogql.identifier(searchTerm)}%'
+                                ORDER BY timestamp DESC
+                                limit 100`,
+                    }
+
+                    const res = await api.query(query)
+
+                    return res.results?.map((x) => x[0]) as string[]
+                },
+            },
+        ],
+
+        topUrls: [
+            null as { url: string; count: number }[] | null,
+            {
+                loadTopUrls: async () => {
+                    const query: HogQLQuery = {
+                        kind: NodeKind.HogQLQuery,
+                        query: hogql`SELECT properties.$current_url AS url, count(1) as count FROM events
```

Comment:
```suggestion
                        query: hogql`SELECT properties.$current_url AS url, count() as count FROM events
```

i don't know if they ignore the `(1)` but clickhouse docs say `count()` has special optimizations

https://clickhouse.com/docs/en/sql-reference/aggregate-functions/reference/count